### PR TITLE
DAOS-10455 test: Enable enable_vmd: true whenever VMD devices exist.

### PR DESCRIPTION
**Pull request by phender**
Set enable_vmd: true in the daos_server yaml if there are VMD devices on
the server hosts regardless of whether or not they are being utilized in
the server config file.  This avoids errors with the server start-up
NVMe scan.

Skip-unit-tests: true
Skip-func-test-vm: true
Skip-func-hw-test-large: true

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

File: src/tests/ftest/launch.py [hunk 1]
```diff
@@ -742,18 +742,19 @@ def get_test_files(test_list, args, yaml_dir, vmd_flag=False):
...
-        yaml_file, env_vars = replace_yaml_file("{}.yaml".format(base), args, yaml_dir, vmd_flag)
+        yaml_file = replace_yaml_file("{}.yaml".format(base), args, yaml_dir, vmd_flag)
         test_file["yaml"] = yaml_file
-        test_file["env"] = extra_env_vars.copy()
-        test_file["env"].update(env_vars)
+
+        # Set enable_vmd: true in the daos_server yaml if there are VMD devices on the host
+        # regardless of whether they are being specified in the server config file to avoid
+        # failures in the server start-up NVMe scan.
+        test_file["env"] = {"DAOS_ENABLE_VMD": "True" if vmd_flag else "False"}
```
Comment (daltonbohning) [inline src/tests/ftest/launch.py:757]: This works too
```suggestion
        test_file["env"] = {"DAOS_ENABLE_VMD": str(vmd_flag)}
```

**Review by daltonbohning — APPROVED** [review]
Changes look simple enough, but does this need to be tested on every cluster "just in case"?

**Review by tanabarr — DISMISSED** [review]

**Comment by phender** [comment]
The PR  was run with one VMD-enabled cluster (HW Medium) and one non-VMD enabled cluster (HW Small).  All of the HW Small tests passed, but there were 57 failures in HW Medium: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8834/4/testReport/

**Review by daltonbohning — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]

File: src/tests/ftest/launch.py [hunk 2]
```diff
@@ -1038,10 +1036,6 @@ def replace_yaml_file(yaml_file, args, yaml_dir, vmd_flag=False):
                     #   0000:81:00.0 --> 0000:12:00.0
                     value_format = "\"{}\""
                     values_to_replace = [value_format.format(item) for item in yaml_find[key]]
-                    # if VMD pci address in present under nvme_data,
-                    # set DAOS_ENABLE_VMD to True
-                    if vmd_flag is True:
-                        env_vars["DAOS_ENABLE_VMD"] = "True"
```
Comment (ashleypittman) [inline src/tests/ftest/launch.py:1044]: This part of the change introduced a pylint error about unused option.

**Pull request merged**